sql/src/main/java/io/crate/analyze/InsertFromSubQueryAnalyzer.java

```diff
@@ -99,4 +105,26 @@ private void validateMatchingColumns(InsertFromSubQueryAnalyzedStatement context
         }
     }
 
+    private void processUpdateAssignments(TableRelation tableRelation,
+                                          InsertFromSubQueryAnalyzedStatement statement,
+                                          List<Assignment> assignments) {
+        Map<Reference, Symbol> updateAssignments = new HashMap<>(assignments.size());
+        for (Assignment assignment : assignments) {
+            Reference columnName = tableRelation.resolveField(
+                    (Field) expressionAnalyzer.convert(assignment.columnName(), expressionAnalysisContext));
+            assert columnName != null;
+
+            Symbol assignmentExpression = expressionAnalyzer.normalizeInputForReference(
```

Comment:
right side must be evaluated with an expressionAnalyzer that has forWrite = false and I think the FieldProvider should also be different.  Depending on wheter  set  tableName.colName = .. is valid in this case or not